tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,231 @@
+from typing import List, Optional, Union
+
+from tamr_unify_client import Client
+from tamr_unify_client.dataset.resource import Dataset
+from tamr_unify_client.attribute.resource import AttributeSpec
+from tamr_unify_client.attribute.type import AttributeType
+
+from tamr_toolbox.models.data_type import JsonDict
+
+
+def create(
+    *,
+    client: Client,
+    dataset_name: str,
+    primary_keys: Optional[List[str]] = None,
+    attributes: Optional[List[str]] = None,
+    attribute_types: Optional[JsonDict] = None,
+    description: Optional[str] = None,
+    external_id: Optional[str] = None,
+    tags: Optional[List[str]] = None,
+    dataset: Optional[Dataset] = None,
+) -> Dataset:
+    """Flexibly create a source dataset in Tamr. Will use array string as default attribute type
+       if none are specified. If a dataset object is passed, a new dataset with dataset_name as
+       its name will be created that has the same attributes and primary keys as the dataset.
+
+    Args:
+        tamr: TUC client
+        dataset_name: name for new dataset
+        primary_keys: one or more attributes for primary key(s) of new dataset
```

Comment:
should this be required or optional? I was thinking you COULD specify it but didn't have to. If you specify them (and their types) it works. Just thinking